scripts/ps/Get-AccessPackage.ps1

```diff
@@ -0,0 +1,393 @@
+}
+
+$groupsDictionary = Get-GroupsDictionary
+$getAllAccessPackagesWithResources = Get-ResourcesFromAccessPackages -GroupDictionary $groupsDictionary
+
+# Get unique group IDs and access package IDs
+$groupIDs = $getAllAccessPackagesWithResources.GroupID | Select-Object -Unique
+$accessPackageIDs = $getAllAccessPackagesWithResources.AccessPackageID | Select-Object -Unique
+
+# Function to Get Entra Role Assignments for Access Package-Linked Groups
+$entraRoles = Get-EntraRolesForGroups -GroupIDs $groupIDs
+
+# Get Group owners from Access package Group ID's above
+$groupOwners = Get-GroupOwnersWithNames -GroupIDs $groupIDs
+
+# Get Access reviewers
+$accessReviewers = Get-AccessReviewGroups -AccessPackageIDs $accessPackageIDs
+
+# --- Merge Everything ---
+
+# Loop through each access package entry and enrich it with related role, owner, and reviewer data
+$mergedExport = foreach ($entry in $getAllAccessPackagesWithResources) {
+
+    # Find any Entra role assignments or eligibilities linked to the group in this access package
+    $matchingRoles = $entraRoles | Where-Object { $_.GroupID -eq $entry.GroupID }
+
+    # Find the owners of the group associated with this access package
+    $matchingOwners = $groupOwners | Where-Object { $_.GroupID -eq $entry.GroupID }
+
+    # Find reviewers assigned to the access package (via access reviews
+    $matchingReviewers = $accessReviewers | Where-Object { $_.AccessPackageID -eq $entry.AccessPackageID }
+
+    # Combine owner names into a single string
+    $ownerNames = ($matchingOwners | Select-Object -ExpandProperty OwnerName) -join ', '
+
+    # Combine reviewer names into a single string
+    $reviewerNames = ($matchingReviewers | Select-Object -ExpandProperty ReviewerGroupName) -join ', '
+    #OLD $reviewerNames = ($matchingReviewers | Select-Object -ExpandProperty ReviewerName) -join ', '
+
+
+    # If no Entra roles are assigned to the group, return a single entry with empty role fields
+    if ($matchingRoles.Count -eq 0) {
+        [PSCustomObject][Ordered]@{
+            Catalog           = $entry.Catalog
+            #CatalogID         = $entry.CatalogID 
+            AccessPackage     = $entry.AccessPackage
+            #AccessPackageID   = $entry.AccessPackageID
+            GroupDisplayname  = $entry.GroupDisplayname
+            #GroupID           = $entry.GroupID
+            RoleName          = "No role found"
+            #RoleDescription   = "No role found"
+            #Scope             = "No role found"
+            AssignmentType    = "no role found"
+            RoleStatus        = "no role found"
+            GroupOwners       = $ownerNames
+            AccessReviewers   = $reviewerNames
+        }
+         # If roles are found, return one entry per role assignment
+    } else {
+        foreach ($role in $matchingRoles) {
+            [PSCustomObject][Ordered]@{
+                Catalog           = $entry.Catalog
+                #CatalogID         = $entry.CatalogID
```

Comment:
Can these commented out properties be removed if not being used or are you planning to add at a later stage?